_component_design/text-input.md

```diff
@@ -0,0 +1,354 @@
+{% endfigure %}
+
+{% endcolumn %}
+
+</div>
+
+</section>
+
+<section class="static-section" markdown="1">
+
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
+
+### Disabled
+
+In a disabled state, a text input cannot be edited by the user and the ``not-allowed`` style is applied to the cursor. The not-allowed cursor style prevents the user from clicking in the text input.
+
+When a field is disabled, consider changing the helper text so the user understands why it is disabled.
+
+{% endcolumn %}
+
+{% column right:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+{% figure [caption:"Text input disabled state"] [class:"image bg-light border"] %}
+<embed src="{{site.url}}/assets/images/components/inputs-and-controls/text-input/textinput-disabled.png" width="400"/>
+{% endfigure %}
+
+{% endcolumn %}
+
+</div>
+
+</section>
+
+<section class="static-section" markdown="1">
+
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
+
+### Error
+
+An inactive and error state occurs when the user closes or submits a form and a required field is empty.
+
+Consider the following guidelines when you construct an error state text input:
+
+- The text input container changes color to reflect the invalid state.
+- Place the error icon to the left of the error message.
+- Place the error message below the helper text.
+- When the user enters a valid value, the text input returns to the enabled state and the error message is removed.
+- Extended error message text extends to the right edge of the text input container.
+
+{% endcolumn %}
+
+{% column right:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+{% figure [caption:"Text input error state"] [class:"image bg-light border"] %}
+<embed src="{{site.url}}/assets/images/components/inputs-and-controls/text-input/textinput-states-error.png" width="428"/>
+{% endfigure %}
+
+{% endcolumn %}
+
+</div>
+
+</section>
+
+<section class="static-section" markdown="1">
+
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-8-md hxSpan-8-lg" %}
+
+## Variations
+
+You can format text inputs to accept only certain types of data. For example, you can add the dollar sign (**$**) prefix to a text input to indicate that the user should enter numeric values.
+
+This section includes specifications for prefix and suffix text inputs, password redaction text inputs, and multi-line text inputs.
+
+{% endcolumn %}
+
+</div>
+
+</section>
+
+<section class="static-section" markdown="1">
+
+<div class="hxRow"  markdown="1">
+
+{% column left:"hxCol hxSpan-12-xs hxSpan-12-sm hxSpan-4-md hxSpan-4-lg" %}
+
+### Prefixes and suffixes
+
+Use a prefix or suffix to clarify the meaning of the numeric input value entered by the user. Left justify a prefix and right justify a suffix.
```

Comment:
```suggestion
Use a prefix or suffix to clarify the meaning of the numeric input value entered by the user. Left align a prefix and right align a suffix.
```
See https://docs.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/l/left-justify-left-justified (which is one of the guides our style guide is based off of).